crates/ruff_linter/src/rules/pydocstyle/rules/indent.rs

```diff
@@ -291,10 +292,58 @@ pub(crate) fn indent(checker: &mut Checker, docstring: &Docstring) {
             let line_indent = leading_space(last);
             let line_indent_size = line_indent.chars().count();
             if line_indent_size > docstring_indent_size {
+                let offset_len = if line_indent.text_len() == last.text_len() {
+                    line_indent_size
+                // The trailing quotes are not on a separate line.
+                } else {
+                    let over_indented_size =
+                        std::cmp::min(line_indent_size - docstring_indent_size, over_indented_size);
+                    match over_indented_size.cmp(&docstring_indent_size) {
```

Comment:
I'm sorry but I'm still confused (about the comments and the code). It can very well be that it's just me not understanding the code.

I'm confused about the comparison of `over_indent_size` with `docstring_indent_size` in the case when `over_indent_size = line_indent_size - docstring_indent_size`. 

```python
def test():
    """
    Abcd
        line"""
```

In this case:

* `line_indent_size`:  8
* `docstring_indent_size`: 4
* outer `over_indented_size`: `usize::MAX`?
* local `over_indented_size: 4 (`8 - 4`)

That means the code now takes the equal branch because `over_indent_size == docstring_indent_size` but this code should be flagged.

Reply:
You are correct that this case should be flagged, but my current approach will overlook it. I am not pretty sure how we can handle both cases if the trailing quotes are on the same line unless we have a way to know the pattern of the previous line. Do you have any idea?

```
def right_indent_quotes_same_line(a: int) -> None:
    """Foo.

    Parameters
    ----------
    a : int
        A parameter."""
    pass
```
```
def test():
    """
    Abcd
        line"""
```